### PR TITLE
improvement/refactor services test

### PR DESCRIPTION
**Pull request by danyecls**

### CODE REVIEW

**Review by otavio — APPROVED** [review]

File: api/services/session_test.go [hunk 1]
```diff
@@ -44,41 +32,56 @@ func TestListSessions(t *testing.T) {
 	cases := []struct {
 		name          string
 		pagination    paginator.Query
-		requiredMocks func()
+		requiredMocks func(query paginator.Query)
```
Comment (henrybarreto) [inline api/services/session_test.go:35]: Should we instantiate the `query` inside the `requiredMocks` as in other tests?

I think I'll enforce the test and make it broken if `query` changes in some way. Am I right?

File: api/services/namespace_test.go [hunk 2]
```diff
@@ -155,59 +218,58 @@ func TestListNamespaces(t *testing.T) {
...
 
 	cases := []struct {
-		name          string
+		description   string
+		user          *models.User
+		namespace     *models.Namespace
 		ctx           context.Context
-		tenantID      string
-		requiredMocks func()
+		requiredMocks func(namespace *models.Namespace, user *models.User)
```
Comment (henrybarreto) [inline api/services/namespace_test.go:235]: As `query`, here it could be the same?

File: api/services/namespace_test.go [hunk 3]
```diff
@@ -296,63 +358,74 @@ func TestCreateNamespace(t *testing.T) {
...
-		},
-		Settings: &models.NamespaceSettings{SessionRecord: true},
-		TenantID: "xxxxx",
-	}
-
 	cases := []struct {
-		name          string
-		requiredMocks func()
+		description   string
+		requiredMocks func(namespace requests.NamespaceCreate)
```
Comment (henrybarreto) [inline api/services/namespace_test.go:363]: here too

File: api/services/namespace_test.go [hunk 4]
```diff
@@ -1075,78 +1512,223 @@ func TestEditNamespaceUser(t *testing.T) {
...
 			},
-			Expected: NewErrNamespaceMemberNotFound(passiveMember.ID, nil),
+			Expected: NewErrNamespaceMemberNotFound("passiveMemberID", nil),
 		},
 		{
-			Name:          "EditNamespaceUser fails when could not find active member inside namespace",
-			TenantID:      namespacePassiveObserver.TenantID,
-			UserID:        activeMember.ID,
-			MemberID:      passiveMember.ID,
+			description:   "EditNamespaceUser fails when could not find active member inside namespace",
```
Comment (henrybarreto) [inline api/services/namespace_test.go:1659]: Should `description` be capitalized like the other fields?

Comment (henrybarreto) [reply api/services/namespace_test.go:1659]: I think you could check to see if there is this inconsistency somewhere else.

File: api/services/session_test.go (hunk 1 above)
Comment (danyecls) [reply api/services/session_test.go:35]: corrected in other cases

File: api/services/namespace_test.go (hunk 3 above)
Comment (danyecls) [reply api/services/namespace_test.go]: corrected in other cases

File: api/services/namespace_test.go (hunk 4 above)
Comment (danyecls) [reply api/services/namespace_test.go]: fixed in all cases

File: api/services/namespace_test.go (hunk 2 above)
Comment (danyecls) [reply api/services/namespace_test.go]: corrected in other cases

**Review by henrybarreto — APPROVED** [review]

**Review by otavio — APPROVED** [review]

**Pull request merged**